calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -216,6 +219,123 @@ def test_validate_arguments(self, case, sys_exit_called):
+        )
+
+        # Set up variables used in assertion statements
+        environment = [
+            "HOSTNAME=%s" % node.hostname,
+            "IP=3.3.3.3",
+            "IP6=%s" % ip6,
+            "ETCD_AUTHORITY=1.1.1.1:80",  # etcd host:port
+            "FELIX_ETCDADDR=1.1.1.1:80",  # etcd host:port
+        ]
+        binds = {
+            "/proc":
+                {
+                    "bind": "/proc_host",
+                    "ro": False
+                },
+            log_dir:
+                {
+                    "bind": "/var/log/calico",
+                    "ro": False
+                },
+            "/run/docker/plugins":
+                {
+                    "bind": "/usr/share/docker/plugins",
+                    "ro": False
+                }
+        }
+
+        # Assert
+        m_os.path.exists.assert_called_once_with('./log_dir')
+        m_check_system.assert_called_once_with(fix=False, quit_if_error=False)
+        m_get_host_ips.assert_called_once_with(4)
+        m_warn_if_unknown_ip.assert_called_once_with('3.3.3.3', ip6)
+        m_warn_if_hostname_conflict.assert_called_once_with('3.3.3.3')
+        m_install_kube.assert_called_once_with(node.KUBERNETES_PLUGIN_DIR)
+        m_client.get_ip_pools.assert_has_calls([call(4), call(6)])
+        m_client.ensure_global_config.assert_called_once_with()
+        m_client.create_host.assert_called_once_with(
+            node.hostname, '3.3.3.3', ip6, as_num
+        )
+        m_docker_client.remove_container.assert_called_once_with(
+            'calico-node', force=True
+        )
+        m_os.getenv.assert_called_once_with(
+            ETCD_AUTHORITY_ENV, ETCD_AUTHORITY_DEFAULT
+        )
+        m_docker.utils.create_host_config.assert_called_once_with(
+            privileged=True,
+            restart_policy={"Name":"Always"},
+            network_mode="host",
+            binds=binds
+        )
+        m_find_or_pull_node_image.assert_called_once_with(
+            'node_image', m_docker_client
+        )
+        m_docker_client.create_container.assert_called_once_with(
+            node_image,
+            name='calico-node',
+            detach=True,
+            environment=environment,
+            host_config='host_config',
+            volumes=['/proc_host',
+                     '/var/log/calico',
+                     '/usr/share/docker/plugins']
+        )
+        m_docker_client.start.assert_called_once_with({'Id':666})
+
+    @patch('calico_ctl.node.client', autospec=True)
+    @patch('calico_ctl.node.docker_client', autospec=True)
+    def test_node_stop(self, m_docker_client, m_client):
+        """
+         Test node_stop function in calicoctl node command
+         """
+        # Call method under test
+        node.node_stop(True)
+
+        # Assert
+        m_client.remove_host.assert_called_once_with(node.hostname)
+        m_docker_client.stop.assert_called_once_with('calico-node')
+
+
```

Comment:
You should add another test here for `node_stop` that raises the `docker.errors.APIError`.  You can do this by creating a subfunction within the test that raises an error and set a `sideeffect` on the mocked out `docker_client.stop` function.  This will cause the `stop` function to run the code you specified in the subfunction.  Then you can assert the function raised the `docker.errors.APIError`.
```
@patches....
def test_node_stop_error(....)
  ...
  def raise_docker_apierror(*args):
    raise docker.errors.APIError
  docker_client.stop.side_effect = raise_docker_apierror
  self.assertRaise(docker.errors.APIError, node_stop, <node_stop_input>)
  ...